datawire/ambassador
# Use Helm for installing AES

## Description

Use the Helm repo for running `edgectl install`

## Related Issues

#2466

## Testing

1 - Basic installation

* install with `edgectl install`
* After a successful installation, check that the Helm release is there:
```console
$ helm list -n ambassador 
NAME      	NAMESPACE 	REVISION	UPDATED                                 	STATUS  	CHART           	APP VERSION
ambassador	ambassador	1       	2020-04-08 14:03:12.893838399 +0200 CEST	deployed	ambassador-6.2.4	1.3.2      
```
* Check that the `app.kubernetes.io/managed-by` label is properly set:
```console
$ kubectl get deployments -n ambassador ambassador -o yaml | grep managed-by
    app.kubernetes.io/managed-by: edgectl
        app.kubernetes.io/managed-by: edgectl
```

2 - Installation after Helm

* Install AES with Helm. For example (in local env, with `NodePort`) with :
```console
$ kubectl create namespace ambassador
namespace/ambassador created
$ helm install ambassador --namespace ambassador datawire/ambassador --set service.type=NodePort --wait
# ...
```
* Check that `edgectl install` detects the Helm installation and continues with the rest of the setup:
```console
 $ ./edgectl install                

Installing the Ambassador Edge Stack

Please enter an email address for us to notify you before your TLS certificate
and domain name expire. In order to acquire the TLS certificate, we share this
email with Let’s Encrypt.
Email address [alvaro.saurin@*******]: 

========================================================================
Beginning Ambassador Edge Stack Installation

-> Checking latest version of Ambassador Edge Stack available...
-> Found an existing installation of Ambassador Edge Stack 1.3.2 [helm].
-> Ambassador was installed with Helm: will continue setup...
-> Checking the AES pod deployment
...
```

3 - Installation after an old Helm chart

* Same as previous scenario. Check that `edgectl install` aborts with a message saying it does not support upgrades.

4 - Installation after applying manifests from `https://www.getambassador.io/yaml/*.yaml`

* Same as previous scenario. Check that `edgectl install` aborts with a message saying it does not support upgrades.
